### ISSUE TITLE
[Feature Request]: merge_swath not smart under certain cases

### ISSUE THREAD

**Issue by Xiaohua-Eric-Xu** [gmtsar/gmtsar#152]
Sometimes merge_swath fail to find the correct mid point when combining data from multiple swaths. A recent fix 5cc6f525c78611637a4d012dc370da7bdfd17bc5 allows the user to input stitching positions (in pixels), but would be great if it is automatic.

**Comment by chintals** [comment]
Hi Dr. Xu, 
I believe the det_stitch code you have added to merge_swath could solve the issue of finding the precise pixel overlap. I have just requested a change to include pixel offset computed in the csh script for a two-subswath merge case.

One odd thing I am still not sure of is the merge_swath c file logs (merge_log) output an n1 pixel offset different from what is computed in the csh file using the det_stitch flag. 

The other question I have is in lines 213 & 215 for merge_swath.c; what is the significance of the constant 150 being used?

Thanks,

**Comment by Xiaohua-Eric-Xu** [comment]
Hi @chintals , it was to just move ~500m from the overlap boundary toward the overlap center. This is an imprecise estimate and there are cases that this may fail. At first, I made the n1, n2 options to overwrite these estimates, and then wrote a det_stitch option to calculate n1, n2 based on the NaN-s surrounding the valid data in the output grid. However, when it has wet region (ocean) in the data, sometimes it may fail. I also noticed that some early data or data from a few specific dates could not be merged together due to the extra large shift in sampling area.

**Comment by Xiaohua-Eric-Xu** [comment]
Hi @chintals, that's right according to the design. However, in reality, the valid data exist as a subsection of the "coverage", and even for data from the same track, sometimes there is no single number that can be used for properly merging. If you look to the history of this number, you'll see that it got changed quite a few times and in the end it stayed at 150 and I turned to making n1/n2 options. A while ago a user from SunYatSen University found this could be predicted by the columns of NaNs on both sides of the grids and then I wrote the det_stitch option based on that.  p.s. burst overlaps should be around 2km-ish.
